rust/kernel/sync/revocable_mutex.rs

```diff
@@ -140,7 +139,7 @@ impl<T: ?Sized> RevocableMutex<T> {
 
         // SAFETY: We know `inner.data` is valid because `is_available` is set to true. We'll drop
         // it here and set it to false so it isn't dropped again.
-        unsafe { drop_in_place(&mut inner.data) };
+        unsafe { ManuallyDrop::drop(&mut inner.data) };
         inner.is_available = false;
```

Comment:
If panicking was implemented using unwinding this would need to be moved above the drop for panic safety. Given that it is instead implemented as aborting in rust-for-linux it doesn't really matter for safety, so I am fine with leaving it as is too. Maybe if someone ports part of the code to the userspace, which may be useful for running tests, it would matter.